log4j-api-java9/src/test/java/org/apache/logging/log4j/util/java9/ServiceLoaderUtilTest.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache license, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the license for the specific language governing permissions and
+ * limitations under the license.
+ */
+package org.apache.logging.log4j.util.java9;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.lang.invoke.MethodHandles;
+import java.util.Collections;
+import java.util.List;
+import java.util.ServiceConfigurationError;
+import java.util.stream.Collectors;
+
+import org.apache.logging.log4j.util.PropertySource;
+import org.apache.logging.log4j.util.ServiceLoaderUtil;
+import org.apache.logging.log4j.util.java9.test.BetterService;
+import org.apache.logging.log4j.util.java9.test.Service;
+import org.junit.jupiter.api.Test;
+
+public class ServiceLoaderUtilTest {
+
+    @Test
+    public void testServiceResolution() {
+        // Run only if we are a module
+        if (ServiceLoaderUtil.class.getModule().isNamed()) {
```

Comment:
You can use `assumeTrue()` from JUnit here.

Reply:
I am afraid I will have to leave it for now, since the module runs tests using Surefire 2.13 and JUnit 3. Upgrading the Surefire plugin is impossible due to [SUREFIRE-2073](https://issues.apache.org/jira/browse/SUREFIRE-2073).

Reply:
Huh, oh well. If this is running via JUnit 3, shouldn't the test extend TestCase or something like that?

Reply:
My mistake, Surefire is running the tests as [POJO tests](https://maven.apache.org/surefire/maven-surefire-plugin/examples/pojo-test.html). Since these also use the `JUnit3Provider`, I thought that JUnit 3 was involved. It isn't.

However these tests are not run by JUnit 5, which wasn't even supported in version 2.13. I'll try some hacks (like compiling the tests using Java 8 and `module-info.java` using Java 9) and maybe it will work on the latest Surefire release.